site/content/concepts/tenancy.md

```diff
@@ -1,12 +1,149 @@
+
+When a device connects to one of Hono's protocol adapters, the adapter determines the tenant this device belongs to.
+How this is done is described for the protocol adapters in the User Guide.
+
+After the tenant is determined, the adapter retrieves the details of the determined tenant by means of the Tenant API.
+Only if the tenant exists and is enabled the adapter further processes the data of the device that is connecting. Otherwise
+the connection will be closed.
+
+## Protocol Adapter Configuration
+
+Protocol adapters retrieve parts of their configuration on a tenant level by using the details of the tenant that was determined.
+
+This includes e.g. if a specific protocol adapter is enabled at all for this tenant, allowing to define tenants with 
+only a subset of Hono's functionality. This feature is foreseen to be especially important for production setups.
+
+An example is to define a tenant that is only enabled for MQTT access, but is not allowed to connect to the HTTP adapter.
+
+Please refer to the [Tenant API]({{< relref "Tenant-API.md" >}}) to find out which protocol adapter properties are
+configurable on a tenant level. The number of such properties may grow during further development of Hono.
+
+## AMQP 1.0 endpoints
+
+The AMQP 1.0 endpoints for all APIs of Hono are scoped to a tenant, by using the scheme `<api>/TENANT/...`.
+
+Examples are:
+
+- `telemetry/TENANT/device`
+- `registration/TENANT/device`
+
+etc.
+
+This separates the AMQP endpoints from each other on a tenant level.
+
+The only exception to this is the [Tenant API]({{< relref "Tenant-API.md" >}}), which does not follow this scheme since it
+is addressing the tenants themselves.   
+
+## Devices and tenants
+
+All devices of Hono belong to exactly one tenant, so any data is sent in the context of the
+tenant a device belongs to.
+
+It is important here to focus on Hono's view of a device: it is on a logical level, meaning a device is created in the 
+context of a tenant purely by using Hono's Credentials and Registration APIs. 
+
+Often a physical device will be created in Hono as exactly one logical device, and thus belongs to exactly one tenant.
+
+But if necessary, this can be extended:  
+
+**Advanced setup for multiple tenants**
+
+Due to the logical view Hono has to devices, a physical device can have several *logical identities* in Hono: if using
+different credentials when connecting to a protocol adapter, the physical device can *decide* which identity to use
+for this connection, and thus can even connect in the context of different tenants.
+
+
+## Upstream data flow for tenants
+
+An important detail in Hono's architecture is that data sent upstream is transported via the tenant
+scoped AMQP 1.0 links from the protocol adapters to the AMQP 1.0 network.
+
+Each tenant has its own pair of AMQP 1.0 links and is treated 
+independently from other tenants regarding the back pressure mechanism that AMQP 1.0 offers (using different credits).
```

Comment:
I would omit the `(using different credits)`. This seems to be  a technical detail that is of not much importance here, is it?